### DISCUSSION TITLE
Service container startup order

### DISCUSSION THREAD

**Discussion by chuckatkins** [community/community#26475] [Actions]
<p>Re: <a href="https://github.com/chuckatkins/miscelaneous-scripts/pull/2" rel="nofollow noopener">https://github.com/chuckatkins/miscelaneous-scripts/pull/2</a></p>
<p>I’ve got a workflow with a container job and a service.  Currently the job container is started first,followed by the service container.  Is there some way to have the service container start first before the job container?  I’m trying to setup jobs to run with emulated CPU architectures so I have the images built for the desired architecture but I need to first run the service that registers the emulation bianries with the kernel before the job container can even be started.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/chuckatkins">@chuckatkins</a> ,</p>
<p>We have no methods to set the startup order of containers and service containers.<br>
If you set set a Docker container for for a job, that means all the steps in this job will be ran in the Docker container, so it must start this Docker container before running the steps. If you also configure service container, the service container will be started but be used until have steps is running to connect it.</p>
<p>Actually, you can see all the containers (regardless of whether they are the service containers) will be automatically created and started in the same pre-step. “Initialize containers”, and we can’t control the startup order of the containers in this pre-step.</p>
<p>More details, you can reference here: <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/about-service-containers#communicating-with-service-containers" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/about-service-containers#communicating-with-service-containers</a></p>
